lib/smart_proxy_dns_powerdns/dns_powerdns_main.rb

```diff
@@ -74,6 +79,18 @@ def domain_id
     end
 
     private
+    def domain
```

Comment:
Yeah, or even look it up and store it in the initialiser as you need it for both create/remove.